app/ota_metadata.py

```diff
@@ -327,28 +356,102 @@ class RegularInf:
     nlink: int
     sha256hash: str
     path: Path
+    _base: str
     size: Optional[int] = None
     inode: Optional[str] = None
 
     _reginf_pa: ClassVar[re.Pattern] = re.compile(
         r"(?P<mode>\d+),(?P<uid>\d+),(?P<gid>\d+),(?P<nlink>\d+),(?P<hash>\w+),'(?P<path>.+)'(,(?P<size>\d+)(,(?P<inode>\d+))?)?"
     )
 
-    def __init__(self, _input: str):
-        _ma = self._reginf_pa.match(_input)
+    @classmethod
+    def parse_reginf(cls, _input: str):
+        _ma = cls._reginf_pa.match(_input)
         assert _ma is not None, f"matching reg_inf failed for {_input}"
 
-        self.mode = int(_ma.group("mode"), 8)
-        self.uid = int(_ma.group("uid"))
-        self.gid = int(_ma.group("gid"))
-        self.nlink = int(_ma.group("nlink"))
-        self.sha256hash = _ma.group("hash")
-        self.path = Path(de_escape(_ma.group("path")))
+        mode = int(_ma.group("mode"), 8)
+        uid = int(_ma.group("uid"))
+        gid = int(_ma.group("gid"))
+        nlink = int(_ma.group("nlink"))
+        sha256hash = _ma.group("hash")
+        path = Path(de_escape(_ma.group("path")))
+
+        # special treatment for /boot folder
+        _base = "/boot" if str(path).startswith("/boot") else "/"
 
-        _size = _ma.group("size")
-        if _size:
-            self.size = int(_size)
+        size, inode = None, None
+        if _size := _ma.group("size"):
+            size = int(_size)
             # ensure that size exists before parsing inode
             # it's OK to skip checking of inode,
             # as un-existed inode will be matched to None
-            self.inode = _ma.group("inode")
+            inode = _ma.group("inode")
+
+        return cls(
+            mode=mode,
+            uid=uid,
+            gid=gid,
+            path=path,
+            nlink=nlink,
+            sha256hash=sha256hash,
+            size=size,
+            inode=inode,
+            _base=_base,
+        )
+
+    def __hash__(self) -> int:
+        """Only use path to distinguish unique RegularInf."""
+        return hash(self.path)
+
+    def __eq__(self, _other) -> bool:
+        """
+        NOTE: also take Path as _other
+        """
+        if isinstance(_other, Path):
+            return self.path == _other
+
+        if isinstance(_other, self.__class__):
+            return self.path == _other.path
+
+        return False
+
+    def make_relative_to_slot(self, slot_mp: Union[Path, str]) -> Path:
```

Comment:
what do you mean by `mp`?
And this class has responsibility for ota metadata operation, so I think it's better not to bring the concept of `slots` in this class.
copy from src to dst ... is OK, but this function doesn't care about `slot`.

Reply:
mp means mount_point.
I think we can rename it to `make_relative_to_mount_point` for better clearance!

Reply:
updated in d46b100.

Reply:
This file still has the name `slot` in the argument name and function name.

Reply:
Oh I will update them all.

Reply:
updated in 3a208eb.